gnomad-browser-infra/main.tf

```diff
@@ -0,0 +1,253 @@
+  account_id   = "${var.infra_prefix}-es-snaps"
+  description  = "The service account for the elasticsearch snapshot lifecycle manager."
+  display_name = "${var.infra_prefix} Elasticsearch Snapshots"
+}
+
+resource "google_service_account" "data_pipeline" {
+  account_id   = "${var.infra_prefix}-data-pipeline"
+  description  = "The service account for runnint the gnomAD data pipeline"
+  display_name = "${var.infra_prefix} Data pipeline Service Account"
+}
+
+resource "google_project_iam_member" "gke_nodes_iam" {
+  for_each = toset([
+    "logging.logWriter",
+    "monitoring.metricWriter",
+    "monitoring.viewer",
+    "stackdriver.resourceMetadata.writer",
+    "storage.objectViewer"
+  ])
+
+  role    = "roles/${each.key}"
+  member  = google_service_account.gke_cluster_sa.member
+  project = var.project_id
+}
+
+resource "google_project_iam_member" "data_pipeline_dataproc_worker" {
+  role    = "roles/dataproc.worker"
+  member  = google_service_account.data_pipeline.member
+  project = var.project_id
+}
+
+# required for requesterPays resources
+resource "google_project_iam_member" "data_pipeline_service_consumer" {
+  role    = "roles/serviceusage.serviceUsageConsumer"
+  member  = google_service_account.data_pipeline.member
+  project = var.project_id
+}
+
+resource "google_service_account_iam_member" "es_snapshots" {
+  role               = "roles/iam.workloadIdentityUser"
+  service_account_id = google_service_account.es_snapshots.name
+  member             = "serviceAccount:${var.project_id}.svc.id.goog[elasticsearch/es-snaps]"
+}
+
+resource "google_storage_bucket_iam_member" "es_snapshots" {
+  bucket = google_storage_bucket.elastic_snapshots.name
+  role   = "roles/storage.admin"
+  member = google_service_account.es_snapshots.member
+}
+
+resource "google_storage_bucket_iam_member" "data_pipeline" {
+  bucket = google_storage_bucket.data_pipeline.name
+  role   = "roles/storage.admin"
+  member = google_service_account.data_pipeline.member
+}
+
+resource "google_storage_bucket" "elastic_snapshots" {
+  name                        = "${var.infra_prefix}-elastic-snaps"
+  location                    = var.es_snapshots_bucket_location
+  storage_class               = "STANDARD"
+  uniform_bucket_level_access = true
+  public_access_prevention    = "enforced"
+
+  labels = {
+    "deployment" = "${var.infra_prefix}"
+    "terraform"  = "true"
+    "component"  = "elasticsearch"
+  }
+}
+
+resource "google_storage_bucket" "data_pipeline" {
+  name                        = "${var.infra_prefix}-data-pipeline"
+  location                    = var.data_pipeline_bucket_location
+  storage_class               = "STANDARD"
+  uniform_bucket_level_access = true
+  public_access_prevention    = "enforced"
+
+  labels = {
+    "deployment" = "${var.infra_prefix}"
+    "terraform"  = "true"
+    "component"  = "data-pipeline"
+  }
+}
+
+# A document containing the Broad's public IP subnets for allowing Office and VPN IPs in firewalls
```

Comment:
So the IP subnets are listed in a file in a google bucket? This is really nice. Will the file path be parameterized? Is it an option to create the infrastructure without the IP/VPN restrictions?

Reply:
That bucket and file are maintained by BITS, and theoretically they're updated regularly/programmatically somehow. The bucket and path shouldn't change, as far as I understand. Having this in here statically definitely makes the module less useful for people other than us, so I think it's probably a good idea to add a toggle for it, and perhaps another input variable that people could use to provide their own subnet ranges.

Reply:
OK, I've updated this so that you can conditionally include the BITS maintained networks list in the master_authorized_networks config (default to false, so that if anyone outside of us uses the module they don't get it added without knowing). I've added a new variable that lets you define your own networks to allow. Empty by default, and we won't typically use that, but it's there for people that need customization.

If the BITS network fetching is enabled, and you also provide a list of authorized nets, both sets of networks are added to the config. This would be useful in the case that we need to supplement the BITS networks with additional networks in the future.